editor/plugins/text_shader_editor.cpp

```diff
@@ -1225,6 +1234,9 @@ TextShaderEditor::TextShaderEditor() {
 	disk_changed->add_button(TTR("Resave"), !DisplayServer::get_singleton()->get_swap_cancel_ok(), "resave");
 	disk_changed->connect("custom_action", callable_mp(this, &TextShaderEditor::save_external_data));
 
+	trim_trailing_whitespace_on_save = EDITOR_GET("text_editor/behavior/files/trim_trailing_whitespace_on_save");
```

Comment:
This line looks like it may have been mistakenly removed in [this PR](https://github.com/godotengine/godot/pull/88474/files#diff-cd71ad8157d2fd2599a3fd5d1944dc81bec04203ba522ba8dbc89a01bdb2064a). Without the initializer getting these values it will mean opening the shader editor won't respect the users settings.

Reply:
I tested it some more and I must have had the settings window open which would have captured the ctrl + S meaning that the trim didn't occur. 

Everything looks good now!